network/test/echoengine_test.go

```diff
@@ -473,6 +473,9 @@ func (suite *EchoEngineTestSuite) singleMessage(echo bool, send ConduitSendWrapp
 		// evaluates content of received message
 		assert.Equal(suite.Suite.T(), event, rcvEvent)
 
+		// evaluates channel that message was received on
+		assert.Equal(suite.Suite.T(), engine.TestNetwork, <-receiver.channel)
```

Comment:
Thanks for applying this comment. Just, we'd better using blocking calls in tests with a timeout. So I would recommend refactoring using a select statement with a timeout. Alternatively, you may develop a variant of [`AssertClosesBefore`](https://github.com/onflow/flow-go/blob/1f7ce61f6a6fbe1f0cbf708de2712ae416533442/utils/unittest/unittest.go#L51-L58) that does an equality assertion when it reads the channel:  

**Refactor**
```
func AssertClosesBeforeEqual(t *testing.T, done <-chan struct{}, expected interface{}, duration time.Duration) {
	select {
	case <-time.After(duration):
		assert.Fail(t, "channel did not return in time")
	case actual := <-done:
		assert.Equal(suite.Suite.T(), expected, actual, "interface equality failed")
	}
}
```

**Usage here**
~`assert.Equal(suite.Suite.T(), engine.TestNetwork, <-receiver.channel)`~
instead
`unittest.AssertClosesBeforeEqual(suite.T(), <-receiver.channel, engine.TestNetwork, 100 * time.Millisecond)`


A **minor comment**: you may use `suite.T()` instead of `suite.Suite.T()`. 
A **general suggestion**: for future PRs with multiple comments, would be great if you could please apply each comment in a separate commit and reply to the comment with the commit link. It makes tracking changes easier. Thanks!

Reply:
What is the difference between `AssertClosesBefore` and [`RequireCloseBefore`](https://github.com/onflow/flow-go/blob/smnzhu%2Fnetwork-multi-channel/utils/unittest/unittest.go#L89)?

Reply:
In `RequireCloseBefore` the rest of the test will not be executed if the channel does not close on time. However, in `AssertClosesBefore` the rest of the test will be executed, even when the channel does not close on time. In both cases, if the channel does not close on time the test eventually fails. So it is a matter of whether we want the rest of the test to be executed, or not.

Reply:
@yhassanzadeh13 so I tried your suggested refactor, and it turns out it will not work because golang does not support passing a specific channel type to a function which accepts a generic channel type. In this example, the `AssertClosesBeforeEqual` function would need to take a channel argument of type `<-chan network.Channel` in order to compile, I cannot pass `receiver.channel` to the function as type `<-chan interface{}` unfortunately...

Reply:
I have a workaround in [5974b96](https://github.com/onflow/flow-go/pull/937/commits/5974b9649dbab2c2db2dc3b62beec6e68d1459a2), take a look and let me know what you think

Reply:
Looks good, thanks for applying this. Though to be DRY maybe you can go with my suggestion on encapsulating the entire logic in a separate test helper. That is fine in either case. I leave it up to you.